content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1056 @@
+        │   ├── post.js
+        │   └── user.js
+        ├── index.js
+        ├── routes.js
+        └── seed.js
+```
+
+The example repository contains a `seed` script inside the `package.json` file. Run `npm run seed` to populate your database with the sample data in the `./src/seed.js` file.
+
+## Step 1. Install the Prisma CLI
+
+The first step to adopt Prisma is to [install the Prisma CLI](/concepts/components/prisma-cli/installation#local-installation-recommended) in your project:
+
+```terminal copy
+npm install prisma --save-dev
+```
+
+## Step 2. Introspect your database
+
+### 2.1. Set up Prisma
+
+Before you can introspect your database, you need to set up your [Prisma schema](/concepts/components/prisma-schema) and connect Prisma to your database. Run the following command in your terminal to create a basic Prisma schema file:
+
+```terminal copy
+npx prisma init --datasource-provider mongodb
+```
+
+This command created:
+
+- A new directory called `prisma` with a `schema.prisma`: Your Prisma schema file that specifies your database connection and models
+- `.env`: A [`dotenv`](https://github.com/motdotla/dotenv) at the root of your project, if it doesn't exist, to configure your database connection URL as an environment variable
+
+The Prisma schema file currently looks as follows:
+
+```prisma file=prisma/schema.prisma
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+datasource db {
+  provider = "mongodb"
+  url      = env("DATABASE_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+```
+
+<Tip>
+
+If you're using VS Code, be sure to install the [Prisma VS Code extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma) for syntax highlighting, formatting, auto-completion and a lot more cool features.
+
+</Tip>
+
+### 2.2. Connect your database
+
+You can configure your [database connection URL](/reference/database-reference/connection-urls#mongodb) in the `.env` file.
+
+The format of the connection URL Mongoose uses is similar to the one Prisma will use.
+
+```bash file=.env
+mongodb://alice:myPassword43@localhost:27017/blog-mongoose
+```
+
+Refer to the [MongoDB connection URL specification](https://www.mongodb.com/docs/manual/reference/connection-string/#connection-string-options) for further details.
+
+### 2.3. Run Prisma's introspection
+
+With your connection URL in place, you can [introspect](/concepts/components/introspection) your database to generate your Prisma models:
+
+> **Note**: MongoDB is a _schemaless_ database. To incrementally adopt Prisma in your project, ensure your database is populated with sample data. Prisma introspects a MongoDB schema by sampling data stored and inferring the schema from the data in the database
+
+```terminal copy
+npx prisma db pull
+```
+
+This creates the following Prisma models:
+
+```prisma file=prisma/schema.prisma
+type UsersProfile {
+  bio String
+}
+
+model categories {
```

Comment:
Are you sure this would not be `Category`? Where does the collection name `categories` come from?

Reply:
`categories` comes from the `Category` schema in Mongoose – the default behaviour is lowercasing and pluralizing collection names